Aquality.Appium.Mobile/src/Aquality.Appium.Mobile/Applications/AqualityServices.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Threading;
+
+namespace Aquality.Appium.Mobile.Applications
+{
+    /// <summary>
+    /// Controls application and Aquality services
+    /// </summary>
+    public class AqualityServices : AqualityServices<IMobileApplication>
+    {
+        private static readonly ThreadLocal<MobileStartup> ApplicationStartupContainer = new ThreadLocal<MobileStartup>(() => new MobileStartup());
+        private static readonly ThreadLocal<IApplicationFactory> ApplicationFactoryContainer = new ThreadLocal<IApplicationFactory>();
+
+        /// <summary>
+        /// Gets registered instance of localized logger
+        /// </summary>
+        public static IApplicationProfile ApplicationProfile => Get<IApplicationProfile>();
+
+        /// <summary>
+        /// Check if application already started.
+        /// </summary>
+        /// <value>true if application started and false otherwise.</value>
+        public new static bool IsApplicationStarted => IsApplicationStarted();
+
+        /// <summary>
+        /// Gets registered instance of localized logger
+        /// </summary>
+        public static ILocalizedLogger LocalizedLogger => Get<ILocalizedLogger>();
+
+        /// <summary>
+        /// Gets registered instance of Logger
+        /// </summary>
+        public static Logger Logger => Get<Logger>();
+
+        /// <summary>
+        /// Gets ConditionalWait object
+        /// </summary>
+        public static IConditionalWait ConditionalWait => Get<IConditionalWait>();
+
+        /// <summary>
+        /// Gets factory to create elements.
+        /// </summary>
+        public static IElementFactory ElementFactory => Get<IElementFactory>();
+
+        /// <summary>
+        /// Resolves required service from <see cref="ServiceProvider"/>
+        /// </summary>
+        /// <typeparam name="T">type of required service</typeparam>
+        /// <exception cref="InvalidOperationException">Thrown if there is no service of the required type.</exception> 
+        /// <returns></returns>
+        public static T Get<T>()
+        {
+            return ServiceProvider.GetRequiredService<T>();
+        }
+
+        /// <summary>
+        /// Provides current instance of application
+        /// </summary>
+        public static IMobileApplication Application
+        {
+            get => GetApplication(StartApplicationFunction, ConfigureServices);
+            set => SetApplication(value);
+        }
+
+        /// <summary>
+        /// Method which allow user to override or add custom services.
+        /// </summary>
+        /// <param name="startup"><see cref="MobileStartup"/>> object with custom or overriden services.</param>
+        public static void SetStartup(MobileStartup startup)
+        {
+            if (startup != null)
+            {
+                ApplicationStartupContainer.Value = startup;
+                SetServiceProvider(ConfigureServices().BuildServiceProvider());
+            }
```

Comment:
should not we throw an exception if someone tries to pass null? it can help to quickly identify what went wrong

Reply:
okay, but I don't remember that we did it in Aquality.Selenium or anywhere else